### PR TITLE
feat: Add password toggle visibility icon.

### LINKED ISSUE #533: Add password toggle visibility icon
**Author the user story for this feature**
As a user, I want to be able to toggle password visibility so I can fix any typing issues.

**Is your feature request related to a problem? Please describe.**
Password mistyping.

**Describe the solution you'd like**
I want to be able to toggle password visibility so I can fix any typing issues. See the designs below
image

**Describe alternatives you've considered**
-

**Additional context**
-

### PR DESCRIPTION
**Pull request by solid-dimakoniaiev**
Fixes #533

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

**Comment by illia-romanenko** [comment]
Designers updated icon - please update the implementation.

File: metrics/web/lib/common/presentation/widgets/metrics_text_form_field.dart [hunk 1]
```diff
@@ -154,7 +140,11 @@ class _MetricsTextFormFieldState extends State<MetricsTextFormField> {
             obscureText: widget.obscureText,
             keyboardType: widget.keyboardType,
             style: textFieldTheme.textStyle,
-            decoration: _decoration,
+            decoration: _decoration.copyWith(
```
Comment (solid-vladlaba) [inline metrics/web/lib/common/presentation/widgets/metrics_text_form_field.dart:143]: I like this! But could you please check that the prefix icon matches the design (for the search field, for example)?

File: metrics/web/test/auth/presentation/widgets/auth_form_test.dart [hunk 2]
```diff
@@ -184,6 +190,41 @@ void main() {
         expect(googleSignInButton.strategy, isA<GoogleSignInOptionStrategy>());
       },
     );
+
+    testWidgets(
+      "applies an eye_on.svg, if the password is obscure",
```
Comment (solid-vladlaba) [inline metrics/web/test/auth/presentation/widgets/auth_form_test.dart:195]:
```suggestion
      "applies the enabled eye icon to the password input field if the password is obscured",
```

File: metrics/web/test/auth/presentation/widgets/auth_form_test.dart [hunk 3]
```diff
@@ -184,6 +190,41 @@ void main() {
...
+
+        final imageWidget = tester.widget<Image>(suffixImageFinder);
+        final networkImage = imageWidget.image as NetworkImage;
+
+        expect(networkImage.url, 'icons/eye_on.svg');
+      },
+    );
+
+    testWidgets(
+      "applies an eye_off.svg, if the password is not obscure",
```
Comment (solid-vladlaba) [inline metrics/web/test/auth/presentation/widgets/auth_form_test.dart:211]:
```suggestion
      "applies the disabled eye icon to the password input field if the password is not obscured",
```

File: metrics/web/test/auth/presentation/widgets/auth_form_test.dart [hunk 4]
```diff
@@ -22,7 +22,13 @@ void main() {
         find.widgetWithText(MetricsTextFormField, AuthStrings.password);
     final submitButtonFinder =
         find.widgetWithText(MetricsPositiveButton, AuthStrings.signIn);
-
+    final suffixImageFinder = find.descendant(
+      of: find.byType(MetricsTextFormField),
```
Comment (solid-vladlaba) [inline metrics/web/test/auth/presentation/widgets/auth_form_test.dart:26]: Should we ensure that the icon set as a suffix icon of the text field? This finder does not ensure that.

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Well done, @solid-dimakoniaiev @solid-yuriiuvarov! I've added a couple of suggestions below - please let me think what do you think!

**Review by solid-vladlaba — APPROVED** [review]
LGTM, thanks @solid-dimakoniaiev @solid-yuriiuvarov !

**Pull request merged**